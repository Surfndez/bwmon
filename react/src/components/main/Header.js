import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { AppBar, Toolbar, Typography } from '@material-ui/core';

const useStyles = makeStyles({
    title: {
        flexGrow: 1,
    },
});

const Header = () => {
    const classes = useStyles();

    return (
        <AppBar position="static">
            <Toolbar variant="dense">
                <Typography id="app-title" variant="h6" className={classes.title}>
                    {process.env.REACT_APP_DESCRIPTION}
                </Typography>
                <Typography id="app-version" className={classes.text}>
                    {process.env.REACT_APP_VERSION}
                </Typography>
            </Toolbar>
        </AppBar>
    );
};

export default Header;
