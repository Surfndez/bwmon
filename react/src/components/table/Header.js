import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableSortLabel from '@material-ui/core/TableSortLabel';

const useStyles = makeStyles((theme) => ({
    visuallyHidden: {
        border: 0,
        clip: 'rect(0 0 0 0)',
        height: 1,
        margin: -1,
        overflow: 'hidden',
        padding: 0,
        position: 'absolute',
        top: 20,
        width: 1,
    },
}));

const Header = ({ prefix, cellInfos, onRequestSort, ascending, orderBy }) => {
    const classes = useStyles();
    const sortHandler = (property) => (event) => {
        onRequestSort(event, property);
    };

    return (
        <TableHead>
            <TableRow key={'header'}>
                {cellInfos.map((cellInfo) => {
                    if (cellInfo.sortable) {
                        const direction = ascending ? 'asc' : 'desc';
                        return (
                            <TableCell
                                id={prefix + '-' + cellInfo.id}
                                key={cellInfo.id}
                                align={cellInfo.align}
                                sortDirection={orderBy === cellInfo.id ? direction : false}
                            >
                                <TableSortLabel
                                    active={orderBy === cellInfo.id}
                                    direction={orderBy === cellInfo.id ? direction : 'asc'}
                                    onClick={sortHandler(cellInfo.id)}
                                >
                                    {cellInfo.label}
                                    {orderBy === cellInfo.id ? (
                                        <span className={classes.visuallyHidden}>
                                            {direction === 'desc' ? 'sorted descending' : 'sorted ascending'}
                                        </span>
                                    ) : null}
                                </TableSortLabel>
                            </TableCell>
                        );
                    } else {
                        return (
                            <TableCell key={cellInfo.id} align={cellInfo.align}>
                                {cellInfo.label}
                            </TableCell>
                        );
                    }
                })}
            </TableRow>
        </TableHead>
    );
};

export default Header;
