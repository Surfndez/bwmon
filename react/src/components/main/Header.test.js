import React from 'react';
import { create } from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {
    test('renders correctly', () => {
        const tree = create(<Header />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
