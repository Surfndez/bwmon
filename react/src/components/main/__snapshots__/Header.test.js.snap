// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
  >
    <h6
      className="MuiTypography-root makeStyles-title-1 MuiTypography-h6"
      id="app-title"
    >
      Bandwidth Monitor Usage
    </h6>
    <p
      className="MuiTypography-root MuiTypography-body1"
      id="app-version"
    >
      3.0.0
    </p>
  </div>
</header>
`;
